Mathlib/Tactic/Linarith/Frontend.lean

```diff
@@ -244,7 +244,7 @@ if it does not succeed at doing this.
 partial def linarith (only_on : Bool) (hyps : List Expr) (cfg : LinarithConfig := {})
     (g : MVarId) : MetaM Unit := do
   -- if the target is an equality, we run `linarith` twice, to prove ≤ and ≥.
-  if (← g.getType).isEq then do
+  if (← instantiateMVars (← g.getType)).isEq then do
```

Comment:
```suggestion
  if (← g.getType').isEq then do
```

Reply:
This doesn't work: `whnf` is pretty violent:
```
0 < (t*(r + v) + s)*3*u
```
turns into `Rat.blt ... < true`...